code/autopilot/autopilot.cpp

```diff
@@ -1023,7 +1023,7 @@ void nav_warp(bool prewarp=false)
 
 				// retime collision pairs
 				if (Objects[Ships[i].objnum].flags[Object::Object_Flags::Collides])
-					obj_collide_retime_cached_pairs(&Objects[Ships[i].objnum]);
+					obj_collide_cache_stale();
```

Comment:
What about marking a single object stale, using an object flag in the same way as `Docked_already_handled` for example?  Then the code could still limit itself to one iteration through the list per frame, but it would check the stale flag in addition to any other checks.  This stale flag would then be cleared after its timestamp is reset.